src/components/layout/tables/AntRow/AntRow.tsx

```diff
@@ -0,0 +1,183 @@
+      );
+    }
+    if (confirmations > 49) {
+      setStatusIcon(
+        <CircleCheck width={20} height={20} fill={'var(--success-green)'} />,
+      );
+    }
+    if (confirmations <= 0) {
+      setStatusIcon(
+        <AlertCircle width={20} height={20} fill={'var(--text-faded)'} />,
+      );
+    }
+  }
+
+  function getName() {
+    if (!antState) {
+      return;
+    }
+    if (antState.name.length > 20) {
+      setNickname(
+        `${antState.name.slice(0, 10)}...${antState.name.slice(-10)}`,
+      );
+    }
+    if (antState.name.length < 20) {
+      setNickname(antState.name);
+    }
+  }
+  function getTargetId() {
+    if (antState.records['@'].transactionId) {
+      setTargetId(antState.records['@'].transaction);
+    }
+    // todo: remove below for v1, this is a legacy format for ant record antStates.
+    if (typeof antState.records['@'] === 'string') {
+      setTargetId(antState.records['@']);
+    }
+  }
+
+  return (
+    <>
+      <tr
+        className="assets-table-header"
+        style={
+          bgColor
+            ? {
+                backgroundColor: bgColor,
+                color: textColor,
+              }
+            : {}
+        }
+      >
+        <>
+          <td
+            className="assets-table-item"
+            style={textColor ? { color: textColor } : {}}
+          >
+            {nickname ? nickname : <Loader size={30} />}
+          </td>
+          <td
+            className="assets-table-item"
+            style={textColor ? { color: textColor } : {}}
+          >
+            {isMobile ? (
+              <CopyTextButton
+                displayText={`${antId.slice(0, 2)}...${antId.slice(-2)}`}
+                copyText={antId}
+                size={24}
+              />
+            ) : (
+              <CopyTextButton
+                displayText={`${antId.slice(0, 6)}...${antId.slice(-6)}`}
+                copyText={antId}
+                size={24}
+              />
+            )}
+          </td>
+          <td
+            className="assets-table-item"
+            style={textColor ? { color: textColor } : {}}
+          >
+            {typeof targetId == 'string' ? (
+              isMobile ? (
+                <CopyTextButton
+                  displayText={`${targetId.slice(0, 2)}...${targetId.slice(
+                    -2,
+                  )}`}
+                  copyText={antId}
+                  size={24}
+                />
+              ) : (
+                <CopyTextButton
+                  displayText={`${targetId.slice(0, 6)}...${targetId.slice(
+                    -6,
+                  )}`}
+                  copyText={antId}
+                  size={24}
+                />
+              )
+            ) : (
+              <Loader size={30} />
+            )}
+          </td>
+          <td
+            className="assets-table-item"
+            style={textColor ? { color: textColor } : {}}
+          >
+            {statusIcon ? (
```

Comment:
do we need to set this in state or can we just have the function return the component to be rendered

e.g.
````
getStatusIcon(antState.confirmations)
```

Reply:
hm. Maybe - i think i have to make it a react component here and render it like <Confirmations />

Reply:
Ok, updated this to its own component called `<TransactionStatus/>` that can be reused elsewhere